commands/flags.go

```diff
@@ -57,23 +67,76 @@ func parseOptions(cliContext *cli.Context) (*options.Options, error) {
 	}
 	logger.SetLevel(level)
 
-	ports := cliContext.IntSlice("port")
-	if len(ports) == 0 {
-		return nil, MissingParam(portFlag.Name)
-	}
-
 	listener := cliContext.String("listener")
 	if listener == "" {
 		return nil, MissingParam(listenerFlag.Name)
 	}
 
+	configFile := cliContext.String("config")
+	if configFile == "" {
+		return nil, MissingParam(configFlag.Name)
+	}
+
+	b, err := loadBytes(configFile)
+	if err != nil {
+		return nil, InvalidConfigFile(configFile)
+	}
+
+	checks, err := parseChecksFromConfig(b)
+	if err != nil {
+		return nil, err
+	}
+
 	return &options.Options{
-		Ports:          ports,
+		Checks:         checks,
 		Listener:       listener,
 		Logger:         logger,
 	}, nil
 }
 
+func parseChecksFromConfig(b []byte) ([]options.Check, error) {
+	var checks Checks
+	var checkSlice []options.Check
+
+	// Use UnmarshalStrict to catch any issues in the config,
+	// such as misspelled keys.
+	err := yaml.UnmarshalStrict(b, &checks)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(checks.TcpChecks) + len(checks.HttpChecks) + len(checks.ScriptChecks) == 0 {
+		return nil, errors.New("no checks found: must specify at least one check")
```

Comment:
"No health checks are defined in your config file <xxx>. Please specify at least one check."